src/eos-updater-apply.c

```diff
@@ -148,6 +152,19 @@ apply (GTask *task,
 
   newbootver = ostree_deployment_get_deployserial (new_deployment);
 
+  if (!eos_extensions_save (data->extensions,
+                            repo,
+                            cancel,
+                            &error))
+    goto error;
+  g_set_object (&data->branch_file, data->extensions->branch_file);
+  g_clear_object (&data->extensions);
```

Comment:
Some of the fields in `EosUpdaterData` are used as a shared data between various stages of the updater. We set the `extensions` field in the poll stage, then later, if apply stage is successful, we save the extensions to disk. 

I'll have to document it.